src/validation.cpp

```diff
@@ -1948,13 +1948,13 @@ int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Para
         ThresholdState state = VersionBitsState(pindexPrev, params, pos, versionbitscache);
         const struct BIP9DeploymentInfo& vbinfo = VersionBitsDeploymentInfo[pos];
         if (vbinfo.check_mn_protocol && state == THRESHOLD_STARTED && !fAssumeMasternodeIsUpgraded) {
-            CScript payee;
+            std::vector<CTxOut> voutMasternodePayments;
             masternode_info_t mnInfo;
-            if (!mnpayments.GetBlockPayee(pindexPrev->nHeight + 1, payee)) {
+            if (!mnpayments.GetBlockTxOuts(pindexPrev->nHeight + 1, 0, voutMasternodePayments)) {
                 // no votes for this block
                 continue;
             }
-            if (!mnodeman.GetMasternodeInfo(payee, mnInfo)) {
+            if (!mnodeman.GetMasternodeInfo(voutMasternodePayments[0].scriptPubKey, mnInfo)) {
```

Comment:
This looks like a temporary shortcut thb which would make little sense after there are many MNs paid in one block. Should probably write this part correctly i.e. "continue _if one of paid masternodes_ is unknown" and "continue if one of masternodes is not upgraded" below.

Reply:
Actually, the idea behind voutMasternodePayments is not to pay multiple masternodes but to split the masternode reward into masternode payment and operator reward payment. Even in the case of masternode shares, a block will still only pay a single masternode but split the reward into multiple recipients.

I added a commit on top which loops through all txouts and if none of them is a masternode payment, it continues.